controlplane/pkg/nsmd/serviceregistry.go

```diff
@@ -69,15 +69,15 @@ func (impl *nsmdServiceRegistry) NewWorkspaceProvider() serviceregistry.Workspac
 	return NewDefaultWorkspaceProvider()
 }
 
-func (impl *nsmdServiceRegistry) RemoteNetworkServiceClient(nsm *registry.NetworkServiceManager) (remote_networkservice.NetworkServiceClient, *grpc.ClientConn, error) {
-	err := tools.WaitForPortAvailable(context.Background(), "tcp", nsm.GetUrl(), 1*time.Second)
+func (impl *nsmdServiceRegistry) RemoteNetworkServiceClient(ctx context.Context, nsm *registry.NetworkServiceManager) (remote_networkservice.NetworkServiceClient, *grpc.ClientConn, error) {
+	err := tools.WaitForPortAvailable(ctx, "tcp", nsm.GetUrl(), 1*time.Second)
```

Comment:
We should have ca entralised way to define these constants. Even better if we cna find some algorithm to calculate them.

Reply:
I've added TODO for WaitForPortAvailable to refactor it.